Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialComponent.cpp

```diff
@@ -451,29 +478,44 @@ namespace AZ
         {
             AzToolsFramework::ScopedUndoBatch undoBatch("Generating materials.");
             SetDirty();
+            
+            Data::AssetId modelAssetId;
+            MeshComponentRequestBus::EventResult(modelAssetId, GetEntityId(), &MeshComponentRequestBus::Events::GetModelAssetId);
 
-            // First generating a unique set of all material asset IDs that will be used for source data generation
-            AZStd::unordered_set<AZ::Data::AssetId> assetIds;
+            EditorMaterialComponentExporter::ExportItemsContainer exportItems;
 
-            auto materialSlots = GetMaterialSlots();
-            for (auto& materialSlotPair : materialSlots)
+            // Generate a list of export items for the set of unique default material assets from the model.
+            for (auto& materialSlotPair : GetMaterialSlots())
             {
-                EditorMaterialComponentSlot* materialSlot = materialSlotPair.second;
-                if (materialSlot->m_id.m_materialAssetId.IsValid())
+                // We only care about material assets that were generated from the model source file, since those are the
+                // ones that would need conversion (other materials already have their own source file). This can be detected
+                // by matching GUID component of the AssetId.
+                Data::AssetId defaultMaterialAssetId = materialSlotPair.second->m_defaultMaterialAsset.GetId();
+                bool materialWasGeneratedFromModel = defaultMaterialAssetId.IsValid() && defaultMaterialAssetId.m_guid == modelAssetId.m_guid;
+                if (materialWasGeneratedFromModel)
```

Comment:
Okay, I could take that approach. It doesn't matter in this changelist because we still always have unique default materials for each slot. In my _next_ commit we'll have the possibility of repeated materials for the defaults, and I could address it at that time where I can test it.

Reply:
Okay, I could look into doing that. It doesn't really matter in this commit because default material assignments are always unique. In my next commit there will be an opportunity for them to reuse the same default, so that would be a good time to nail this down.

Reply:
Fair enough.